### PR TITLE
[Flang] Add new Integration tests directory to Flang

### PR DESCRIPTION
**Pull request by TIFitis**
As per the RFC: [https://discourse.llvm.org/t/rfc-flang-new-directory-for-adding-end-to-end-tests-for-lowering-to-llvm-ir-in-flang/74872/11](https://discourse.llvm.org/t/rfc-flang-new-directory-for-adding-end-to-end-tests-for-lowering-to-llvm-ir-in-flang/74872/11), this patch adds a new Integration test directory for OpenMP- `flang/test/Integration/OpenMP` and moves the existing OpenMP integration tests from `flang/test/Driver/OpenMP` to this directory.

**This directory can be used to add Integration tests involving multiple stages of the compiler (for eg. from Fortran to LLVM IR). It should not contain executable tests. We should only add tests here sparingly and only if there is no other way to test. Repeat this message in each test that is added to this directory and sub-directories.**

### CODE REVIEW

**Review by banach-space — APPROVED** [review]
LGTM, thanks for driving this!

Please wait for other reviewers to take a look before landing this.

**Comment by joker-eph** [comment]
Thanks for the CC, I'm not qualified to review though, it's very dependent on what the Flang maintainer think they want to maintain, and how much this should grow.

From a MLIR point of view, we are supporting Flang and we'll support these tests, we may have to revisit this if this gets out-of-hand (hard to maintain / fragile), but this is very hypothetical.

And thanks for caring about better testing for the project!

**Review by kiranchandramohan — APPROVED** [review]
LGTM subject to the following. Please add a pointer to this patch in the RFC.

Since you have not added a CMake flag to enable the tests in this directory, it is important to convey that tests here should be used only under particular circumstances.
Please add a Readme file in the flang/test/Integration directory saying that "This directory can be used to add Integration tests involving multiple stages of the compiler (for eg. from Fortran to LLVM IR). It should not contain executable tests. We should only add tests here sparingly and only if there is no other way to test. Repeat this message in each test that is added to this directory and sub-directories."

**Comment by TIFitis** [comment]
> LGTM subject to the following. Please add a pointer to this patch in the RFC.
> 
> Since you have not added a CMake flag to enable the tests in this directory, it is important to convey that tests here should be used only under particular circumstances.

Thanks for the review, I have added the message to these places.

**Pull request merged**
